certbot-dns-rfc2136/certbot_dns_rfc2136/dns_rfc2136.py

```diff
@@ -0,0 +1,157 @@
+"""DNS Authenticator using RFC 2136 Dynamic Updates."""
+import logging
+
+import dns.flags
+import dns.message
+import dns.name
+import dns.query
+import dns.rdataclass
+import dns.rdatatype
+import dns.tsig
+import dns.tsigkeyring
+import dns.update
+import zope.interface
+
+from certbot import errors
+from certbot import interfaces
+from certbot.plugins import dns_common
+
+logger = logging.getLogger(__name__)
+
+
+@zope.interface.implementer(interfaces.IAuthenticator)
+@zope.interface.provider(interfaces.IPluginFactory)
+class Authenticator(dns_common.DNSAuthenticator):
+    """DNS Authenticator using RFC 2136 Dynamic Updates
+
+    This Authenticator uses RFC 2136 Dynamic Updates to fulfull a dns-01 challenge.
+    """
+
+    description = 'Obtain certs using a DNS TXT record (if you are using BIND for DNS).'
+    ttl = 120
+
+    def __init__(self, *args, **kwargs):
+        super(Authenticator, self).__init__(*args, **kwargs)
+        self.credentials = None
+
+    @classmethod
+    def add_parser_arguments(cls, add):
+        super(Authenticator, cls).add_parser_arguments(add)
+        add('credentials', help='RFC 2136 credentials INI file.')
+
+    def more_info(self):
+        return 'This plugin configures a DNS TXT record to respond to a dns-01 challenge using ' + \
+               'RFC 2136 Dynamic Updates.'
+
+    def _setup_credentials(self):
+        self.credetials = self._configure_credentials(
+            'credentials',
+            'RFC 2136 credentials INI file',
+            {
+                'name': 'TSIG key name',
+                'secret': 'TSIG key secret',
+                'algorithm': 'TSIG key algorithm',
```

Comment:
If `algorithm` is optional, you probably don't want to include it here. But you may want to include some logic later in `_setup_credentials` to ensure the value is valid if set.